src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1034,9 +1034,11 @@ def _validate_admin_password(password, os_type):
     is_linux = (os_type.lower() == 'linux')
     max_length = 72 if is_linux else 123
     min_length = 12
+    error_msg = ("Rule 1: The password length must be between {} and {}\n"
+                 "Rule 2: Password must have the 3 of the following: 1 lower case character, "
+                 "1 upper case character, 1 number and 1 special character").format(min_length, max_length)
     if len(password) not in range(min_length, max_length + 1):
-        raise CLIError('The password length must be between {} and {}'.format(min_length,
-                                                                              max_length))
+        raise CLIError("Your password is invalid for it violates Rule 1\n{}".format(error_msg))
```

Comment:
LGTM. why not make it simpler. error_msg = Password length must be between {} and {}, and must have the 3 of following....". In CLIError, just message "Your password doesn't meet requirement, pls see password constraint: {}".format(error_msg)

Reply:
I think user knows where is error better when we split the error cases.

Reply:
Maybe we can just show the rule that the user violates instead of printing all rules.